### PR TITLE
[17.0][FIX] edi_oca: error in component

### PR DESCRIPTION
**Pull request by manuelregidor**
The component that shows the buttons on the top of the invoices form view is not properly migrated, which is the cause of an error that prevents the access to invoices. This fix sorts out the issue.

T-6447

### CODE REVIEW

**Comment by manuelregidor** [comment]
@ValentinVinagre Could you review, please?

**Review by ValentinVinagre — APPROVED** [review]
LGTM 👍🏻

**Comment by etobella** [comment]
Where it was failing? Seeing the JS it shouldn't...

**Comment by ValentinVinagre** [comment]
@manuelregidor Councillor, could you add some screenshots with the error given by the migration of the "l10n_es_facturae_face" module without this fix? 
Ty 😃

**Comment by manuelregidor** [comment]
@etobella 

We have identified this issue when migrating the module l10n_es_facturae_face. When trying to access an invoice whose customer has the value "FACe" set in the field l10n_es_facturae_sending_code, a frontend error occurs.

edi_oca_error_1

Complete error:

```
OwlError: An error occured in the owl lifecycle (see this Error's "cause" property)
    Error: An error occured in the owl lifecycle (see this Error's "cause" property)
        at handleError (http://localhost:17069/web/assets/00a5a52/web.assets_web.min.js:916:101)
        at App.handleError (http://localhost:17069/web/assets/00a5a52/web.assets_web.min.js:1545:29)
        at Fiber._render (http://localhost:17069/web/assets/00a5a52/web.assets_web.min.js:941:19)
        at Fiber.render (http://localhost:17069/web/assets/00a5a52/web.assets_web.min.js:939:6)
        at ComponentNode.initiateRender (http://localhost:17069/web/assets/00a5a52/web.assets_web.min.js:1007:47)

Caused by: TypeError: Cannot read properties of undefined (reading 'name')
    at Field.template (eval at compile (http://localhost:17069/web/assets/00a5a52/web.assets_web.min.js:1503:374), <anonymous>:16:92)
    at Fiber._render (http://localhost:17069/web/assets/00a5a52/web.assets_web.min.js:940:96)
    at Fiber.render (http://localhost:17069/web/assets/00a5a52/web.assets_web.min.js:939:6)
    at ComponentNode.initiateRender (http://localhost:17069/web/assets/00a5a52/web.assets_web.min.js:1007:47)
```

This error is solved by the modifying the file /static/src/js/widget_edi.esm.js. The following lines

```
EdiConfigurationWidget.template = "edi_oca.EdiConfigurationWidget";

registry.category("fields").add("edi_configuration", EdiConfigurationWidget);
```

need to be replaced by

```
EdiConfigurationWidget.template = "edi_oca.EdiConfigurationWidget";

registry.category("fields").add("edi_configuration", {
    component: EdiConfigurationWidget,
});
```

It is not enough with those changes though, as another error occurs:

edi_oca_error_2

Detailed error:

```
OwlError: Invalid loop expression: "undefined" is not iterable
    Error: Invalid loop expression: "undefined" is not iterable
        at prepareList (http://localhost:17069/web/assets/83382b6/web.assets_web.min.js:1120:12)
        at EdiConfigurationWidget.template (eval at compile (http://localhost:17069/web/assets/83382b6/web.assets_web.min.js:1503:374), <anonymous>:14:54)
        at Fiber._render (http://localhost:17069/web/assets/83382b6/web.assets_web.min.js:940:96)
        at Fiber.render (http://localhost:17069/web/assets/83382b6/web.assets_web.min.js:939:6)
        at ComponentNode.initiateRender (http://localhost:17069/web/assets/83382b6/web.assets_web.min.js:1007:47)
```

It is because the component template is not properly defined. That is why the following function has been added to /static/src/js/widget_edi.esm.js file.

```
    getInfo() {
        return this.props.record.data.edi_config
    }
```
The way data is retrieved from the template is also modified. These lines

```
        <t t-foreach="props.value" t-as="rule_id" t-key="rule_index">
            <t t-set="rule" t-value="props.value[rule_id]" />
```

has been replaced by the following:

```
        <t t-foreach="this.getInfo()" t-as="rule_id" t-key="rule_index">
            <t t-set="rule" t-value="this.getInfo()[rule_id]" />
```

**Comment by ValentinVinagre** [comment]
@peluko00 could u review? 
https://github.com/OCA/edi-framework/pull/91

File: edi_oca/static/src/js/widget_edi.esm.js [hunk 1]
```diff
@@ -23,8 +24,14 @@ export class EdiConfigurationWidget extends Component {
         );
         this.action.doAction(action);
     }
+
+    getInfo() {
```
Comment (etobella) [inline edi_oca/static/src/js/widget_edi.esm.js]:
```suggestion
    get info() {
```

File: edi_oca/static/src/js/widget_edi.esm.js [hunk 2]
```diff
@@ -23,8 +24,14 @@ export class EdiConfigurationWidget extends Component {
...
+        return this.props.record.data.edi_config;
```
Comment (etobella) [inline edi_oca/static/src/js/widget_edi.esm.js]:
```suggestion
        return this.props.record.data[this.props.name];
```
This way it uses the field

File: edi_oca/static/src/js/widget_edi.esm.js [hunk 3]
```diff
@@ -11,6 +11,7 @@ export class EdiConfigurationWidget extends Component {
         this.orm = useService("orm");
         this.action = useService("action");
     }
+
```
Comment (etobella) [inline edi_oca/static/src/js/widget_edi.esm.js]: It is not necessary

File: edi_oca/static/src/xml/widget_edi.xml [hunk 4]
```diff
@@ -1,8 +1,8 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 <templates id="template" xml:space="preserve">
     <div t-name="edi_oca.EdiConfigurationWidget">
-        <t t-foreach="props.value" t-as="rule_id" t-key="rule_index">
-            <t t-set="rule" t-value="props.value[rule_id]" />
+        <t t-foreach="this.getInfo()" t-as="rule_id" t-key="rule_index">
```
Comment (etobella) [inline edi_oca/static/src/xml/widget_edi.xml]:
```suggestion
        <t t-foreach="info" t-as="rule_id" t-key="rule_index">
```

File: edi_oca/static/src/xml/widget_edi.xml [hunk 5]
```diff
@@ -1,8 +1,8 @@
...
+            <t t-set="rule" t-value="this.getInfo()[rule_id]" />
```
Comment (etobella) [inline edi_oca/static/src/xml/widget_edi.xml]:
```suggestion
            <t t-set="rule" t-value="info[rule_id]" />
```

**Review by etobella — CHANGES_REQUESTED** [review]

File: edi_oca/static/src/js/widget_edi.esm.js (hunk 1 above)
Comment (etobella) [reply edi_oca/static/src/js/widget_edi.esm.js]: if you want to have a getInfo, maybe it is better to have a get funcion, isn't it :wink:

**Comment by manuelregidor** [comment]
@etobella Thanks for your feedback. Changes applied.

**Review by etobella — APPROVED** [review]
Thanks, it is cleaner now :smile:

**Pull request merged**